### DISCUSSION TITLE
sourcekit-lsp setup on mac

### DISCUSSION THREAD

**Discussion by Tassil0** [LazyVim/LazyVim#908] [Q&A] [answered]
I have trouble setting up `sourcekit-lsp` on my mac. It is installed. I tried have this in my plugin config:

```
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        -- pyright will be automatically installed with mason and loaded with lspconfig
        pyright = {},
        sourcekit = {
          cmd = { "xcrun", "sourcekit-lsp" },
        },
      },
    },
  },
```

But when I open a .swift file and run `:LspInfo` I have zero clients attached to the current buffer.

**Comment by dpetka2001** [comment]
Not on a Mac but on Linux. Downloaded the Swift toolchain and then put your configuration exactly but with a small change since I'm on Linux, `cmd = { /full/path/to/"sourcekit-lsp" }` because it wasn't on my PATH and didn't want to add it just for testing this. If it's already on your PATH there's no need for full path. Then just made a dir and `cd`'ed into the directory. Then `swift package init`, which created a structure and when I opened a file in the `Sources` dir, the `sourcekit` LSP server attached correctly. The LSP server to attach it looks for a `Package.swift` file or `.git` if it's inside a repo.

**Reply by Tassil0** [reply]
I have the lsp in my path, I tried specifying the full path just to be sure, but no luck. I have both `Package.swift` and `.git` in my repo.

**Reply by dpetka2001** [reply]
What does `LspInfo` show? Does Neovim recognize the filetype of the file as `swift`?

**Reply by Tassil0** [reply]
LspInfo output:
```
 Language client log: /Users/tassilo/.local/state/nvim/lsp.log
 Detected filetype:   swift
 
 0 client(s) attached to this buffer: 
 
 Configured servers list: html, pyright, tailwindcss, dockerls, astro, tsserver, eslint, emmet_ls, clangd, rust_analyzer, graphql, jsonls, lua_ls
```

**Reply by amaanq** [reply]
you didn't configure it, try adding it to the settings table of lspconfig's opts

**Reply by Tassil0** [reply]
@amaanq I think I did that? Please refer to my plugin config in the OP.

**Reply by amaanq** [reply]
you put that in servers, not settings

Hmm nvm disregard, now I'm a bit lost too

**Reply by amaanq** [reply]
Trying on a random swift repo w/ my config here has the lsp running fwiw:
repo - https://github.com/buresdv/Mlem 
config - https://github.com/amaanq/nvim-config/blob/master/lua/plugins/extras/lang/swift.lua

**Reply by Tassil0** [reply]
@amaanq tried replicating your config and explicitly setting filetypes, but again no luck and I get the same behaviour. I even tried it on the Mlem repo

**Reply by amaanq** [reply]
I genuinely have no clue then, maybe you didn't include the file w/ the relevant code for sourcekit or it's not installed

**Reply by Tassil0** [reply]
@amaanq Tried cloning your entire config and only `copilot` attaches to my swift buffer. I can confirm `sourcekit-lsp -h` outputs help correctly, so it should be installed.

**Reply by amaanq** [reply]
Well my config doesn't include the swift extras by default fyi (I practically never use it), so include that in config/lazy.lua otherwise idk

**Reply by Tassil0** [reply]
I can't find `sourcekit` in Mason. Could that be an issue?

**Reply by Tassil0** [reply, marked as answer]
I found the issue... my lsp config was in `example.lua` from the default install, so it never got included. Thank you both for your time and help. Now I feel really stupid :D

**Reply by amaanq** [reply]
It happens :laughing:
